auth.go

```diff
@@ -0,0 +1,58 @@
+package mesosAuthPlugin
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+)
+
+// pathLogin (the function) returns the "login" path struct. It is a function
+// rather than a method because we never call it once the backend struct is
+// built and we don't want name collisions with any request handler methods.
+func pathLogin(b *mesosBackend) *framework.Path {
+	return &framework.Path{
+		Pattern: "login",
+		Fields: map[string]*framework.FieldSchema{
+			"task-id": &framework.FieldSchema{
+				Type: framework.TypeString,
+			},
+		},
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.UpdateOperation: b.pathLogin,
+		},
+	}
+}
+
+// pathLogin (the method) is the "login" path request handler.
+func (b *mesosBackend) pathLogin(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+
+	taskID := d.Get("task-id").(string)
+	if len(taskID) < 1 {
```

Comment:
`< 1` seems weird?

Reply:
We want a non-empty string, and I haven't yet confirmed that `len()` can never return a negative number...

This will almost certainly be replaced with more sensible validation logic once we have something to check the task id against.

Reply:
It seems that checking `len(s) != 0` or `s != ""` are the standards.

Reply:
But this is fine for now if it's about to completely change.